### ISSUE TITLE
Formalize YAML ModSpec approach

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/metanorma-ogc#493]
In OGC GeoPose, ISO 19115-3 and ISO 19135, we have implemented a Liquid template to convert ModSpect instances from YAML to presentation.

This involves 2 Liquid templates:
* Requirements class + Provision / Requirement / Recommendation / Permission
* Conformance class + Conformance test

We wish to formalize this approach for users (ModSpec users including OGC and ISO/TC 211) to use.

The implementation of this can be found here:
* OGC GeoPose: https://github.com/metanorma/ogc-GeoPose/tree/main/standard/standard/modspec
* ISO 19115-3: https://github.com/metanorma/iso-19115-3/tree/main/sources/ed2/sections/tables
* ISO 19135: https://github.com/metanorma/iso-19135/tree/main/sources/iso-19135-2021/sections/templates

We will also need to update documentation on metanorma.org.

This would greatly assist our current work with the MUDDI SWG.

**Comment by opoudjis** [comment]
This is an interim YAML solution, which we can document, pending the machine readable modspec of #242

**Comment by ghobona** [comment]
Discussed 2024-02-15

**Comment by opoudjis** [comment]
@manuelfuenmayor Could you possibly write a blog post documenting the usage of YAML in modspec? If not, I'll do so, but you are more familiar with it than I am, I like to think :)

**Comment by opoudjis** [comment]
I will investigate whether there are YAML schemas that we can exploit in this formalisation.

**Comment by opoudjis** [comment]
https://www.npmjs.com/package/pajv

It's in Node, and it's an extension of the horrific JSON Schema validator ajv.

If this yields unacceptable obscure validation responses, or if we we can always convert the YAML to ruby hashes, and validate them programmatically.

**Comment by ronaldtse** [comment]
@opoudjis we use JSON/YAML schemas for Glossarist already so we should continue the practice.

**Comment by opoudjis** [comment]
Right, useful to know. OK, then it will be voxpupuli, when whoever it is defines the JSON Schema for Modspec.

**Comment by ronaldtse** [comment]
FYI I've made a first cut here, serializable to YAML/JSON/XML schema:
* https://github.com/metanorma/modspec-ruby

**Comment by manuelfuenmayor** [comment]
> FYI I've made a first cut here, serializable to YAML/JSON/XML schema:

Hm, perhaps this affects the validity of the blog post I've written?

**Comment by ronaldtse** [comment]
> Hm, perhaps this affects the validity of the [blog post](https://github.com/metanorma/metanorma.org/pull/758) I've written?

The post has been merged. Thanks!

**Comment by ronaldtse** [comment]
The next to do is to create a Metanorma/LutaML plugin that uses the `modspec` gem. It already provides the JSON/YAML/XML Schemas.

**Comment by ghobona** [comment]
Adding the Blog posts's link for future reference. 

https://www.metanorma.org/blog/2024-04-04-ogc-modspec-in-yaml/

**Comment by ronaldtse** [comment]
The integration ticket is:
* https://github.com/metanorma/modspec-ruby/issues/2

**Comment by opoudjis** [comment]
Get Lutaml-Model to generate a schema for this.

**Comment by ronaldtse** [comment]
This is the same functionality for #242 .

**Comment by ghobona** [comment]
@ronaldtse To have a look at https://www.ogc.org/standards/modspec/

**Comment by ronaldtse** [comment]
The latest ModSpec XML has been published, I will investigate to see how we can facilitate:
* ModSpec YAML <=> XML
* Metanorma => ModSpec XML
